py/dml/toplevel.py

```diff
@@ -357,7 +358,10 @@ def parse_main_file(inputfilename, explicit_import_path, strict):
     version = site.dml_version()
     dml.globals.dml_version = version
     version_str = fmt_version(version)
-    dml.globals.compat_dml12 = version == (1, 2) and not strict
+    if version != (1, 2):
+        for feature in [compat.dml12_inline, compat.dml12_not,
+                        compat.dml12_misc]:
+            dml.globals.enabled_compat.discard(feature)
 
```

Comment:
It feels weird this is done separately from all other compatibility enablement logic. Though we want to do such logic as early as possible, and we have no choice to place _this_ part of it right here.

Reply:
Oh, and I notice that I forgot dml12_goto there! By sheer luck, this is not a problem since `goto` is forbidden by the grammar.